app/Http/Controllers/Resources/BookingController.php

```diff
@@ -17,34 +17,39 @@ class BookingController extends Controller
     public function index(Request $request)
     {
         $query = Booking::with(['eventService' => function ($query) {
-                $query->select('id', 'service_provider_id');
-            }, 'user' => function ($query) {
-                $query->select('id', 'is_blocked');
-            }]);
+            $query->select('id', 'service_provider_id');
+        }, 'user' => function ($query) {
+            $query->select('id', 'is_blocked');
+        }]);
         $query->orderBy('booking_date');
 
         $allowedStatuses = ['pending', 'confirmed', 'canceled', 'completed'];  // Define the allowed statuses
 
         if (auth()->user()->usertype === "service_provider") {
-         // Check if the request has a valid status and filter based on the allowed statuses
+            // Check if the request has a valid status and filter based on the allowed statuses
             if ($request->has('status') && in_array($request->input('status'), $allowedStatuses)) {
                 $query->where('status', $request->input('status'));
             } else {
                 // Redirect back with an error message if the status is invalid
                 return redirect()->route('service-provider.bookings.index', ['status' => 'pending'])
-                                ->with('error', 'Invalid status. Allowed statuses are: pending, confirmed, or canceled.');
+                    ->with('error', 'Invalid status. Allowed statuses are: pending, confirmed, or canceled.');
             }
 
-            $bookings = $query->get();
-            $nonBlockedBookings = $bookings->where('user.is_blocked', false);
-            $blockedBookings = $bookings->where('user.is_blocked', true);
+            // Paginate non-blocked bookings
+            $nonBlockedBookings = $query->clone()->whereHas('user', function ($query) {
+                $query->where('is_blocked', false);
+            })->paginate(6);
```

Comment:
use this to make the status query remain when paginating
`->paginate(6)->appends(['status' => $status]);`